beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/BlockOperationSelectorFactory.java

```diff
@@ -149,14 +150,31 @@ public Consumer<BeaconBlockBodyBuilder> createSelector(
                         .getExecutionPayloadHeaderSchema()
                         .getHeaderOfDefaultPayload(),
                 (executionPayloadContext) -> {
+                  Optional<BuilderForcedFallbackReason> fallbackReason = Optional.empty();
+
                   final boolean transitionNotFinalized =
                       executionPayloadContext
                           .getForkChoiceState()
                           .getFinalizedExecutionBlockHash()
                           .isZero();
 
+                  if (transitionNotFinalized) {
+                    fallbackReason =
+                        Optional.of(BuilderForcedFallbackReason.TRANSITION_NOT_FINALIZED);
+                  } else {
+                    final int window = 32;
+                    final int missingBlockThreshold = 10;
```

Comment:
assuming a couple of CLI args here with sensible limits will feed these two values...